test/TxAtomicSimple.t.sol

```diff
@@ -0,0 +1,217 @@
+        return handlePacket(p);
+    }
+
+    function extHandleAcknowledgement(Packet calldata p, bytes calldata ackBytes) external {
+        handleAcknowledgement(p, ackBytes);
+    }
+
+    function extHandleTimeout(Packet calldata p) external {
+        handleTimeout(p);
+    }
+
+    function setModule(IContractModule m) external {
+        _module = m;
+    }
+}
+
+/// -------- Tests --------
+
+contract TxAtomicSimpleTest is Test {
+    DummyHandler private handler;
+    TestableTxAtomicSimple private harness;
+
+    event OnContractCall(bytes indexed txId, uint8 indexed txIndex, bool indexed success, bytes ret);
+
+    function setUp() public {
+        handler = new DummyHandler();
+        harness = new TestableTxAtomicSimple(IIBCHandler(address(handler)));
+    }
+
+    /// ---- Helpers ----
+
+    function _mkPacketWithCall(bytes memory txId, bytes memory callInfo) internal pure returns (Packet memory p) {
+        Any.Data memory emptyAny = Any.Data({type_url: "", value: ""});
+        ReturnValue.Data memory emptyRet = ReturnValue.Data({value: ""});
+
+        AuthAccount.Data[] memory signers;
+        Any.Data[] memory objects;
+
+        PacketDataCallResolvedContractTransaction.Data memory txResolved =
+            PacketDataCallResolvedContractTransaction.Data({
+                cross_chain_channel: emptyAny,
+                signers: signers,
+                call_info: callInfo,
+                return_value: emptyRet,
+                objects: objects
+            });
+
+        PacketDataCall.Data memory callData = PacketDataCall.Data({tx_id: txId, tx: txResolved});
+
+        bytes memory anyPayload = Any.encode(
+            // solhint-disable-next-line gas-small-strings
+            Any.Data({type_url: "/cross.core.atomic.simple.PacketDataCall", value: PacketDataCall.encode(callData)})
+        );
+
+        HeaderField.Data[] memory fields;
+        bytes memory packetDataBytes =
+            PacketData.encode(PacketData.Data({header: Header.Data({fields: fields}), payload: anyPayload}));
+
+        p.data = packetDataBytes;
+    }
+
+    function _decodeAckStatus(bytes memory ack) internal pure returns (PacketAcknowledgementCall.CommitStatus) {
+        Acknowledgement.Data memory ackData = Acknowledgement.decode(ack);
+        PacketData.Data memory pd = PacketData.decode(ackData.result);
+        Any.Data memory any_ = Any.decode(pd.payload);
+        PacketAcknowledgementCall.Data memory pac = PacketAcknowledgementCall.decode(any_.value);
+        return pac.status;
+    }
+
+    function test_handlePacket_Success_ReturnsOK_AndEmitsEvent() public {
+        bytes memory ret = hex"010203";
+        harness.setModule(new SuccessModule(ret));
+
+        bytes memory txId = hex"deadbeef";
+        bytes memory callInfo = hex"c0ffee";
+        Packet memory packet = _mkPacketWithCall(txId, callInfo);
+
+        // assert event emitted
+        vm.expectEmit(true, true, true, true);
```

Comment:
Since all arguments are always set to `true`, it’s simpler and better to just write `vm.expectEmit(address(harness))`. In addition, testing the emitter address is important.